libr/core/canal.c

```diff
@@ -1154,26 +1154,28 @@ static int core_anal_graph_nodes(RCore *core, RAnalFunction *fcn, int opts) {
 	} else if (is_json) {
 		// TODO: show vars, refs and xrefs
 		char *fcn_name_escaped = r_str_escape_utf8_for_json (fcn->name, -1);
-		r_cons_printf ("{\"name\":\"%s\"", r_str_get (fcn_name_escaped));
+		pj_o (pj);
+		pj_ks (pj, "name", r_str_get (fcn_name_escaped));
 		free (fcn_name_escaped);
-		r_cons_printf (",\"offset\":%"PFMT64d, fcn->addr);
-		r_cons_printf (",\"ninstr\":%"PFMT64d, (ut64)fcn->ninstr);
-		r_cons_printf (",\"nargs\":%d",
+		pj_kn (pj, "offset", fcn->addr);
+		pj_ki (pj, "ninstr", fcn->ninstr);
+		pj_ki (pj, "nargs",
 			r_anal_var_count (core->anal, fcn, 'r', 1) +
 			r_anal_var_count (core->anal, fcn, 's', 1) +
 			r_anal_var_count (core->anal, fcn, 'b', 1));
-		r_cons_printf (",\"nlocals\":%d",
+		pj_ki (pj, "nlocals",
 			r_anal_var_count (core->anal, fcn, 'r', 0) +
 			r_anal_var_count (core->anal, fcn, 's', 0) +
 			r_anal_var_count (core->anal, fcn, 'b', 0));
-		r_cons_printf (",\"size\":%d", r_anal_fcn_size (fcn));
-		r_cons_printf (",\"stack\":%d", fcn->maxstack);
-		r_cons_printf (",\"type\":\"%s\"", r_anal_fcn_type_tostring (fcn->type));
-		//r_cons_printf (",\"cc\":%d", fcn->call); // TODO: calling convention
+		pj_kn (pj, "size",  r_anal_fcn_size (fcn));
+		pj_ki (pj, "stack", fcn->maxstack);
```

Comment:
i dont see why maxstack should be negative, this is probably a task to discuss in a different pr.

you are missing some casts to (st64) because r_anal_fcn_size returns ut32